### ISSUE TITLE
Payment status flow is not timely when call cancel progress.

### ISSUE THREAD

**Issue by kunxiang-chowbus** [stripe/stripe-terminal-ios#175]
## Code to reproduce
![image](https://user-images.githubusercontent.com/64571263/197977401-6f6a696e-c7dc-4fbf-bcae-df3a1eec3b82.png)

## iOS version
iOS 14

## Installation method
Pods

## SDK version
2.10.0

## Other information
When I click on the cancel process, and the sdk callback has been successful, but when I initiate a new payment request, an error (busy error) will be reported. I try to get payment status, I get status = .notReady.

中文描述：
当我点击cancel支付进程，并且SDK已经回调成功了，但是当我发起新的支付请求时，会报错误（busy error），大概是busy，因为症正在执行cancel payment操作。我尝试获取payment status，得到的status = .notReady

**Comment by bric-stripe** [comment]
Hi @kunxiang-chowbus I think I reproduced what you're reporting but want to confirm:

1. collectPaymentMethod, which returns a cancelable
2. cancel that cancelable
3. in the completion block for the _cancel_ request the payment status is not ready and attempts to collect again error with Busy

is that correct?

If so that's currently how the SDK is working and the cancel completion will be called before the operation that was canceled has fully cleaned up[^1]. The guidance would be to only attempt to send another command in the command that was canceled completion block. The command's completion block is the source of truth for the command being fully completed (either successfull, error'd or canceled).

1. collectPaymentMethod, which returns a cancelable
2. cancel that cancelable
3. the cancel's completion block is called and can note the error in case need to handle cancel not succeeding
4. if canceled, the collectPayment completion block should now be called with a `SCPErrorCanceled`. At this time the paymentStatus should be ready and a request for a new command should succeed

This snippet edited from our example app is what outlines it as well:

```swift
        self.cancelable = Terminal.shared.collectPaymentMethod(intent, collectConfig: self.collectConfig) { intentWithPaymentMethod, attachError in
            self.cancelable = nil
            if let error = attachError {
                print(">>> Should be ready: \(Terminal.shared.paymentStatus)")
                self.collectPaymentMethod(intent: intent)
            } else if let intent = intentWithPaymentMethod {
                self.confirmPaymentIntent(intent: intent)
            }
        }
        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
            self.cancelable?.cancel({ error in
                print(">>> Not yet ready: \(Terminal.shared.paymentStatus)")
                // Attempts to Terminal.shared.collect again or anything will error with busy
            })
        }
```

[^1]: I took note of this for future improvements as I can see how this is a bit unexpected and ideally it seems the operation should be fully wrapped up before the cancel's completion block is called.
